docs/design/coreclr/jit/hot-cold-splitting.md

```diff
@@ -0,0 +1,173 @@
+# Hot/Cold Code Splitting
+
+This document describes the current state of hot/cold splitting in the JIT.
+
+Hot/Cold splitting is an optimization where generated code is split into frequently-executed ("hot") and
+rarely-executed ("cold") parts and placed in separate memory regions. Increased hot code density better leverages
+spatial locality, which can improve application performance via fewer instruction cache misses, less OS paging, and
+fewer TLB misses.
+
+Hot/Cold splitting in the JIT was previously implemented for AOT-compiled NGEN images in .NET Framework. With hot/cold
+splitting support in Crossgen2 [in progress](https://github.com/dotnet/runtimelab/tree/feature/hot-cold-splitting)
+(and no existing support for splitting dynamically-generated code), the feature has gone untested in the JIT since
+retiring .NET Framework -- as such, regressions have surely been introduced. Furthermore, this existing implementation
+has several major limitations; functions with certain features, like exception handling or switch tables, cannot be
+split yet. Finally, with ARM64/x64 performance parity being a high priority for CoreCLR, it is key hot/cold splitting
+is implemented for ARM64 code generation, too.
```

Comment:
I think somebody could get confused by the mixed use of tense here and not be entirely sure what the "current state" actually is. 

Consider sectioning this part off as `## Background` and describing just the before state in past tense.